From foo@baz.org Sun Jun  5 14:33:21 2005
Date: Mon, 20 Jun 2005 21:15:16 -0700
To: Greg KH <gregkh@suse.de>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: devfs: Remove devfs_*_tape() functions from the kernel tree

Removes the devfs_register_tape() and devfs_unregister_tape() functions
and all callers of them.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/char/viotape.c          |    3 ---
 drivers/ide/ide-tape.c          |    3 +--
 drivers/scsi/osst.c             |    3 +--
 drivers/scsi/st.c               |    3 +--
 include/linux/devfs_fs_kernel.h |    7 -------
 5 files changed, 3 insertions(+), 16 deletions(-)

--- gregkh-2.6.orig/include/linux/devfs_fs_kernel.h
+++ gregkh-2.6/include/linux/devfs_fs_kernel.h
@@ -26,11 +26,4 @@ static inline int devfs_mk_dir(const cha
 static inline void devfs_remove(const char *fmt, ...)
 {
 }
-static inline int devfs_register_tape(const char *name)
-{
-	return -1;
-}
-static inline void devfs_unregister_tape(int num)
-{
-}
 #endif				/*  _LINUX_DEVFS_FS_KERNEL_H  */
--- gregkh-2.6.orig/drivers/scsi/st.c
+++ gregkh-2.6/drivers/scsi/st.c
@@ -4069,7 +4069,7 @@ static int st_probe(struct device *dev)
 			      S_IFCHR | S_IRUGO | S_IWUGO,
 			      "%s/mt%sn", SDp->devfs_name, st_formats[i]);
 	}
-	disk->number = devfs_register_tape(SDp->devfs_name);
+	disk->number = -1;
 
 	sdev_printk(KERN_WARNING, SDp,
 		    "Attached scsi tape %s", tape_name(tpnt));
@@ -4124,7 +4124,6 @@ static int st_remove(struct device *dev)
 			scsi_tapes[i] = NULL;
 			st_nr_dev--;
 			write_unlock(&st_dev_arr_lock);
-			devfs_unregister_tape(tpnt->disk->number);
 			sysfs_remove_link(&tpnt->device->sdev_gendev.kobj,
 					  "tape");
 			for (mode = 0; mode < ST_NBR_MODES; ++mode) {
--- gregkh-2.6.orig/drivers/char/viotape.c
+++ gregkh-2.6/drivers/char/viotape.c
@@ -246,7 +246,6 @@ static struct device *tape_device[VIOTAP
  */
 static struct {
 	unsigned char	cur_part;
-	int		dev_handle;
 	unsigned char	part_stat_rwi[MAX_PARTITIONS];
 } state[VIOTAPE_MAX_TAPE];
 
@@ -964,7 +963,6 @@ static int viotape_probe(struct vio_dev 
 	devfs_mk_cdev(MKDEV(VIOTAPE_MAJOR, i | 0x80),
 			S_IFCHR | S_IRUSR | S_IWUSR, "iseries/nvt%d", i);
 	sprintf(tapename, "iseries/vt%d", i);
-	state[i].dev_handle = devfs_register_tape(tapename);
 	printk(VIOTAPE_KERN_INFO "tape %s is iSeries "
 			"resource %10.10s type %4.4s, model %3.3s\n",
 			tapename, viotape_unitinfo[i].rsrcname,
@@ -978,7 +976,6 @@ static int viotape_remove(struct vio_dev
 
 	devfs_remove("iseries/nvt%d", i);
 	devfs_remove("iseries/vt%d", i);
-	devfs_unregister_tape(state[i].dev_handle);
 	class_device_destroy(tape_class, MKDEV(VIOTAPE_MAJOR, i | 0x80));
 	class_device_destroy(tape_class, MKDEV(VIOTAPE_MAJOR, i));
 	return 0;
--- gregkh-2.6.orig/drivers/ide/ide-tape.c
+++ gregkh-2.6/drivers/ide/ide-tape.c
@@ -4709,7 +4709,6 @@ static void ide_tape_release(struct kref
 			MKDEV(IDETAPE_MAJOR, tape->minor + 128));
 	devfs_remove("%s/mt", drive->devfs_name);
 	devfs_remove("%s/mtn", drive->devfs_name);
-	devfs_unregister_tape(g->number);
 	idetape_devs[tape->minor] = NULL;
 	g->private_data = NULL;
 	put_disk(g);
@@ -4890,7 +4889,7 @@ static int ide_tape_probe(ide_drive_t *d
 			S_IFCHR | S_IRUGO | S_IWUGO,
 			"%s/mtn", drive->devfs_name);
 
-	g->number = devfs_register_tape(drive->devfs_name);
+	g->number = -1;
 	g->fops = &idetape_block_ops;
 	ide_register_region(g);
 
--- gregkh-2.6.orig/drivers/scsi/osst.c
+++ gregkh-2.6/drivers/scsi/osst.c
@@ -5814,7 +5814,7 @@ static int osst_probe(struct device *dev
 				S_IFCHR | S_IRUGO | S_IWUGO,
 				"%s/ot%sn", SDp->devfs_name, osst_formats[mode]);
 	}
-	drive->number = devfs_register_tape(SDp->devfs_name);
+	drive->number = -1;
 
 	sdev_printk(KERN_INFO, SDp,
 		"osst :I: Attached OnStream %.5s tape as %s\n",
@@ -5846,7 +5846,6 @@ static int osst_remove(struct device *de
 				devfs_remove("%s/ot%s", SDp->devfs_name, osst_formats[mode]);
 				devfs_remove("%s/ot%sn", SDp->devfs_name, osst_formats[mode]);
 			}
-			devfs_unregister_tape(tpnt->drive->number);
 			put_disk(tpnt->drive);
 			os_scsi_tapes[i] = NULL;
 			osst_nr_dev--;
